core/store/src/main/java/org/locationtech/geowave/core/store/base/BaseDataStore.java

```diff
@@ -499,10 +439,23 @@ private boolean isAllAdapters(
 		}
 		if (((query.getQueryConstraints() == null) || (query.getQueryConstraints() instanceof EverythingQuery))) {
 			if ((query.getDataTypeQueryOptions().getTypeNames() == null)
-					|| (query.getDataTypeQueryOptions().getTypeNames().length == 0) || isAllAdapters(
-							query.getDataTypeQueryOptions().getTypeNames())) {
+					|| (query.getDataTypeQueryOptions().getTypeNames().length == 0)
+					|| isAllAdapters(query.getDataTypeQueryOptions().getTypeNames())) {
 				// TODO what about authorizations here?
-				return deleteEverything();
+				// for now just delete all of the data itself and leave stats,
+				// indices, adapters, etc. alone
+				try {
+					// delete the data and the stats for all adapter index/pairs
```

Comment:
This logic could just fall into the logic at the bottom of this method (actually iterating is how stats get updated so it is required).

Reply:
I think the first `if` statement prevents the logic from fully being rolled into logic at the bottom, but I added the iteration.